docs/semgrep-ci/configuration-reference.md

```diff
@@ -19,23 +19,33 @@ While environment variables are the preferred way to configure Semgrep CI, pass
 SEMGREP_RULES="p/security-audit p/secrets"
 ```
 
-## Suppressing blocking findings or errors
+## Configuring blocking findings or errors
 
-Most CI providers block pull requests (PRs) or merge requests (MRs) on non-zero exit codes. Semgrep exits with a non-zero exit code (exit code 1) when it reports a blocking finding or there is an issue in running Semgrep (exit code 2 and above). For more information about specific Semgrep exit codes see [CLI reference](../../cli-reference/#exit-codes).
+Most CI providers block pull requests (PRs) or merge requests (MRs) when the pipeline ends with an exit code other than zero (other than status successful). However, Semgrep in CI is by default configured to suppress internal errors and only report blocking findings to stop your pipeline. Configure this behavior by the `--suppress-errors` or `--no-suppress-errors` flags.
 
-However, Semgrep in CI can run both in fail open and fail closed states. Configure the behavior of Semgrep in CI to suppress blocking findings or internal Semgrep errors blocking your pipeline by using the following options in your YAML configuration file: 
+In further detail, the default behavior of Semgrep in CI towards blocking findings and errors is as follows:
+- When the default behavior is configured, Semgrep encounters an error (exit code `2` and above) pipeline is **not** blocked. Semgrep in CI sends a crash report to a crash-reporting server and lets CI job continue. This report is anonymous and contains information that is usually sent to Semgrep App.
+- A pipeline is blocked when Semgrep reports a blocking finding (exit code `1`). Blocking findings can mean the following:
+    - Findings defined in [Rule Board](https://semgrep.dev/orgs/-/board) of Semgrep App. To avoid blocking findings, remove rules from the **Block** column of [Rule Board](https://semgrep.dev/orgs/-/board).
+    - If you do **not** use Semgrep App with Semgrep in CI, blocking findings encompass **all** Semgrep findings.
 
-- `semgrep ci` - Semgrep in CI **fails** on blocking findings, CI **fails** on internal errors.
-- `semgrep ci || [ $? != 1 ]` - Semgrep in CI **fails** on blocking findings, CI **passes** on internal errors.
+Configure and change the setup of blocking findings or errors in your CI pipeline by using the following options in your YAML configuration file:
+
+- `semgrep ci` - This is the default state. Semgrep in CI **fails** on blocking findings, CI **passes** on internal errors. Semgrep sends an anonymous crash report to a crash-reporting server, and then exits with exit code `0`. Optional: Define it explicitly by using `--suppress-errors` flag.
+- `semgrep ci --no-suppress-errors` - Semgrep in CI **fails** on blocking findings, CI **fails** on internal errors. If you use this flag, all exit codes, including internal errors, are surfaced to the CI provider.
```

Comment:
@spencerdrak Just a heads up - what are we trying to say by "surfaced to the CI provider"? Displayed to CI provider? Or displayed in the interface of the CI provider?

```suggestion
- `semgrep ci --no-suppress-errors` - Semgrep in CI **fails** on blocking findings, CI **fails** on internal errors. If you use this flag, all exit codes, including Semgrep internal errors, are surfaced to the CI provider.
```

Reply:
The CI provider is reading the error code of each command it runs, and if one of these commands returns something other than 0, the build with fail and the CI provider will show the Red X. If you fail-open, and semgrep suppresses and error, then these errors aren't reported to the CI provider since the actual exit code is 0.

Reply:
Thank you!